projects/ziti-console-lib/src/lib/features/visualizer/identity-service-path/identity-service-path.component.ts

```diff
@@ -86,8 +86,12 @@ export class IdentityServicePathComponent implements OnInit {
     this.hostConfigApiUrl = `${this.controllerDomain}/edge/management/v1/configs`;
     this.dialPolicyApiUrl = `${this.controllerDomain}/edge/management/v1/service-policies`;
     this.bindPolicyApiUrl = `${this.controllerDomain}/edge/management/v1/service-policies`;
-    this.getEndpointNetworkAsCode();
 
+    try {
+       this.getEndpointNetworkAsCode();
+    } catch (err:any) {
+      console.log(err);
+    }
```

Comment:
Generally you want to limit the use of try/catch blocks unless absolutely necessary. Ideally we would use more conditional (if/else) to prevent the code paths that result in the errors. Also optional chaining can be used to prevent errors like `Cannot read properties of undefined`

Reply:
agreed. let me use optional chaining.